cli/cook/cli.py

```diff
@@ -0,0 +1,339 @@
+        try:
+            resp = fetch_jobs_on_cluster(cluster, uuids)
+            if resp.status_code == 200:
+                jobs = resp.json()
+                if pred and not pred(jobs):
+                    jobs = wait_until(lambda: jobs_satisfy_pred(cluster, uuids), timeout, interval)
+                    if not jobs:
+                        raise Exception('Timeout waiting for jobs')
+                all_jobs.extend(jobs)
+                for job in jobs:
+                    uuids.remove(uuid.UUID(job['uuid']))
+                if len(uuids) == 0:
+                    return all_jobs
+        except requests.exceptions.ConnectionError as ce:
+            logging.info(ce)
+    raise Exception('Unable to fetch job(s) on the following cluster(s): %s' % clusters)
+
+
+def all_jobs_completed(jobs):
+    """Returns jobs if they are all completed, otherwise False."""
+    if not [job for job in jobs if job.get('status') != 'completed']:
+        return jobs
+    else:
+        return False
+
+
+def wait(clusters, args):
+    """Waits for job(s) with the given UUID(s) to complete."""
+    timeout = args.get('timeout')
+    interval = args.get('interval')
+    fetch_jobs(clusters, args, all_jobs_completed, timeout, interval)
+
+
+actions.update({'wait': wait})
+
+###################################################################################################
+
+show_parser = subparsers.add_parser('show', help='show job(s) by uuid')
+show_parser.add_argument('uuid', nargs='+')
+show_parser.add_argument('--json', help='show the job(s) in JSON format', dest='json', action='store_true')
+
+
+def millis_to_timedelta(ms):
+    """Converts milliseconds to a timedelta for display on screen"""
+    return 'MAX_LONG' if ms == sys.maxsize else datetime.timedelta(milliseconds=ms)
+
+
+def millis_to_date_string(ms):
+    """Converts milliseconds to a date string for display on screen"""
+    s, ms = divmod(ms, 1000)
+    string = '%s.%03d' % (time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(s)), ms)
+    return string
+
+
+def format_timestamps(instance):
+    """Given an instance, formats the timestamps"""
+    instance['end_time'] = millis_to_date_string(instance['end_time'])
+    instance['start_time'] = millis_to_date_string(instance['start_time'])
+    instance['mesos_start_time'] = millis_to_date_string(instance['mesos_start_time'])
+    return instance
+
+
+def tabulate_job(job):
+    """Given a job, returns a string containing tables for the job and instance fields"""
+    headers = ['Field', 'Value']
+    instances = job.pop('instances')
+    job['max_runtime'] = millis_to_timedelta(job['max_runtime'])
+    job['submit_time'] = millis_to_date_string(job['submit_time'])
+    job_table = tabulate(sorted(job.items()), headers=headers)
+    instance_table = tabulate([format_timestamps(i) for i in instances], headers='keys')
+    return 'Job:\n\n%s\n\nInstances:\n\n%s' % (job_table, instance_table)
+
+
+def show(clusters, args):
```

Comment:
Perhaps we should make another command to display the instances, or make something like `--instances` a command line option, without which you do not display detailed instance data.